### PR TITLE
Clarify that exemplar reservoir default may change in a minor version

### PR DESCRIPTION
**Pull request by jack-berg**
Related to #3756. 

See this [conversation](https://github.com/open-telemetry/opentelemetry-specification/pull/3870#discussion_r1483140692) for context.

### CODE REVIEW

**Comment by jack-berg** [comment]
cc @jsuereth / @reyang

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -1052,6 +1052,12 @@ use `AlignedHistogramBucketExemplarReservoir`.
   twenty (e.g. `min(20, max_buckets)`).
 - All other aggregations will use `SimpleFixedSizeExemplarReservoir`.
 
+Exemplar default reservoirs MAY change in
+a [minor version bump](./../versioning-and-stability.md#minor-version-bump).
+Exemplars are considered "best effort". Changing the default reservoir doesn't
```
Comment (jsuereth) [inline specification/metrics/sdk.md:1057]: Another way to phrase this:

We make no guarantees on the shape or statistical properties of returned exemplars.

**Review by jsuereth — APPROVED** [review]

**Comment by CodeBlanch** [comment]
I have a question about this:

> Exemplar default reservoirs MAY change in a minor version bump. No guarantees are made on the shape or statistical properties of returned exemplars.

The spec has pretty clear defaults defined: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exemplar-defaults

@cijothomas Is recommending that in .NET we do a design that is this (more or less):

>Exemplar defaults
>The SDK will include three types of built-in exemplar reservoirs:
>
>`SimpleFixedSizeExemplarReservoir`
>`AlignedHistogramBucketExemplarReservoir`
>`NoopExemplarReservoir`
>
>By default:
>
>All metrics will use the `NoopExemplarReservoir` by default. In order to capture `Exemplar`s users MUST use the view API to select either `SimpleFixedSizeExemplarReservoir`, `AlignedHistogramBucketExemplarReservoir`, or a custom `ExemplarReservoir`.

This seems to be a big deviation from the spec switching `Exemplar`s to be individually opt-in. My question is, was the change introduced on this PR intended to give SDKs such leeway?

**Comment by jack-berg** [comment]
No, the intent of this change was to allow us to later introduce new spec-defined reservoirs, and be able to change the default reservoir for a particular instrument type. In particular, there are likely improvements possible to the reservoir strategy for exponential bucket histograms.

**Pull request merged**